Frameworks/Foundation/NSData.mm

```diff
@@ -80,49 +80,56 @@ - (NSString*)base64EncodedStringWithOptions:(NSDataBase64EncodingOptions)options
     Wrappers::HString encodedString;
     RETURN_NULL_IF_FAILED(cryptographicBufferStatics->EncodeToBase64String(wrlBuffer.Get(), encodedString.GetAddressOf()));
 
-    unsigned int rawLength;
-    const wchar_t* rawEncodedString = WindowsGetStringRawBuffer(encodedString.Get(), &rawLength);
-
-    NSString* nsEncodedString = [NSString _stringWithHSTRING:encodedString.Get()];
-
     // Do any necessary post-processing for options
     // If line length specified, place a newline character (/r, /n, or /r/n every (specified number) characters)
     if ((options & NSDataBase64Encoding64CharacterLineLength) || (options & NSDataBase64Encoding76CharacterLineLength)) {
         size_t lineLength = (options & NSDataBase64Encoding64CharacterLineLength) ? 64 : 76;
         // Calculate number of times to insert newlines
         // Since no newline is needed at the end if stringLength is evenly divisible by lineLength,
         // subtract 1 from stringLength when dividing
-        size_t stringLength = [nsEncodedString length];
-        size_t newLineCount = (stringLength != 0) ? ((stringLength - 1) / lineLength) : 0;
+        unsigned int rawLength;
+        const wchar_t* rawEncodedString = WindowsGetStringRawBuffer(encodedString.Get(), &rawLength);
+
+        size_t newLineCount = (rawLength != 0) ? ((rawLength - 1) / lineLength) : 0;
 
         if (newLineCount > 0) {
             NSDataBase64EncodingOptions useCR = options & NSDataBase64EncodingEndLineWithCarriageReturn;
             NSDataBase64EncodingOptions useLF = options & NSDataBase64EncodingEndLineWithLineFeed;
 
-            StrongId<NSString> newLineChar;
+            const wchar_t* newLineChar = L"\r\n";
             if (useCR && !useLF) {
-                newLineChar = @"\r";
+                newLineChar = L"\r";
             } else if (!useCR && useLF) {
-                newLineChar = @"\n";
-            } else {
-                // Use CRLF by default, but can also be manually specified
-                newLineChar = @"\r\n";
+                newLineChar = L"\n";
             }
 
-            NSMutableString* mutableEncodedString = [NSMutableString stringWithString:nsEncodedString];
+            const size_t newLineLength = wcslen(newLineChar);
+            const size_t newLength = rawLength + newLineCount * newLineLength;
 
-            // Add new line every (lineLength) chars
-            // Iterate backwards to avoid compensating for changes in length
-            for (size_t i = newLineCount; i > 0; i--) {
-                [mutableEncodedString insertString:newLineChar atIndex:(i * lineLength)];
+            std::unique_ptr<wchar_t> result(new wchar_t[newLength]);
+
+            const wchar_t* source = rawEncodedString;
+            wchar_t* dest = result.get();
+
+            for (size_t i = 0; i < newLineCount; ++i) {
+                memcpy(dest, source, lineLength * sizeof(wchar_t));
```

Comment:
I rewrote that part using std::wstring, it was still 3x slower than this version.  Since this is not overly complicated, I am leaving in the better performing code.